processor/src/main/java/org/treblereel/gwt/xml/mapper/apt/processor/BeanProcessor.java

```diff
@@ -90,7 +90,10 @@ private void processBean(TypeElement bean) {
 
   private void processField(VariableElement field) {
     if (checkField(field)) {
-      checkTypeAndAdd(field.asType());
+      // Ensure the serializer/deserializer is generated for the fields concrete type, considering
```

Comment:
Don't assume the field's type is correct.. try to look it up from annotations too.